google-cloud-tools-plugin/ultimate/src/com/google/cloud/tools/intellij/appengine/server/instance/AppEngineServerModel.java

```diff
@@ -270,6 +282,10 @@ public void setAdminHost(String adminHost) {
 
   @Override
   public Integer getAdminPort() {
+    Integer adminPort = settings.getAdminPort();
+    if (adminPort == null) {
```

Comment:
I wasn't quite sure how to hook into that strategy.  getDefaultPort() isn't called anywhere in our code, so I assumed that the JEE IJ support framework was handling that.